pkg/build/rpm_test.go

```diff
@@ -9,123 +10,194 @@ import (
+		file2Path := filepath.Join(rpmSourceDir, "rpm2.rpm")
+		file2, err = os.Create(file2Path)
+		require.NoError(t, err)
+	}
 
-	file2Path := filepath.Join(rpmSourceDir, "rpm2.rpm")
-	defer os.Remove(file2Path)
-	file2, err := os.Create(file2Path)
-	require.NoError(t, err)
+	return tmpDir, rpmSourceDir, func() {
+		if addFiles {
+			assert.NoError(t, file1.Close())
+			assert.NoError(t, file2.Close())
+		}
+		assert.NoError(t, os.RemoveAll(tmpDir))
+	}
+}
+
+func TestGetRPMFileNames(t *testing.T) {
+	// Setup
+	_, rpmSourceDir, teardown := setupRPMSourceDir(t, true)
+	defer teardown()
 
 	// Test
-	rpmFileNames, err := builder.getRPMFileNames(rpmSourceDir)
+	rpmFileNames, err := getRPMFileNames(rpmSourceDir)
 
 	// Verify
 	require.NoError(t, err)
 
 	assert.Contains(t, rpmFileNames, "rpm1.rpm")
 	assert.Contains(t, rpmFileNames, "rpm2.rpm")
-
-	// Cleanup
-	assert.NoError(t, file1.Close())
-	assert.NoError(t, file2.Close())
 }
 
 func TestCopyRPMs(t *testing.T) {
 	// Setup
-	context, err := NewContext("../config/testdata", "", true)
+	tmpDir, rpmSourceDir, teardown := setupRPMSourceDir(t, true)
+	defer teardown()
+
+	tmpDestDir := filepath.Join(tmpDir, "temp-dest-dir")
+	err := os.Mkdir(tmpDestDir, 0o755)
 	require.NoError(t, err)
-	defer func() {
-		assert.NoError(t, CleanUpBuildDir(context))
-	}()
 
-	builder := &Builder{
-		context: context,
-	}
+	// Test
+	err = copyRPMs(rpmSourceDir, tmpDestDir, []string{"rpm1.rpm", "rpm2.rpm"})
+
+	// Verify
+	require.NoError(t, err)
 
-	rpmSourceDir := filepath.Join(context.ImageConfigDir, "rpms")
+	_, err = os.Stat(filepath.Join(tmpDestDir, "rpm1.rpm"))
+	require.NoError(t, err)
 
-	file1Path := filepath.Join(rpmSourceDir, "rpm1.rpm")
-	defer os.Remove(file1Path)
-	file1, err := os.Create(file1Path)
+	_, err = os.Stat(filepath.Join(tmpDestDir, "rpm2.rpm"))
 	require.NoError(t, err)
+}
+
+func TestGetRPMFileNamesNoRPMs(t *testing.T) {
+	// Setup
+	_, rpmSourceDir, teardown := setupRPMSourceDir(t, false)
+	defer teardown()
+
+	// Test
+	rpmFileNames, err := getRPMFileNames(rpmSourceDir)
+
+	// Verify
+	require.ErrorContains(t, err, "no RPMs found")
+
+	assert.Empty(t, rpmFileNames)
+}
+
+func TestCopyRPMsNoRPMDestDir(t *testing.T) {
+	// Setup
+	_, rpmSourceDir, teardown := setupRPMSourceDir(t, true)
+	defer teardown()
+
+	// Test
+	err := copyRPMs(rpmSourceDir, "", []string{"rpm1.rpm", "rpm2.rpm"})
+
+	// Verify
+	require.ErrorContains(t, err, "RPM destination directory cannot be empty")
+}
+
+func TestCopyRPMsNoRPMSrcDir(t *testing.T) {
+	// Setup
+	tmpDestDir, _, teardown := setupRPMSourceDir(t, true)
+	defer teardown()
+
+	// Test
+	err := copyRPMs("", tmpDestDir, []string{"rpm1.rpm", "rpm2.rpm"})
 
-	file2Path := filepath.Join(rpmSourceDir, "rpm2.rpm")
-	defer os.Remove(file2Path)
-	file2, err := os.Create(file2Path)
+	// Verify
+	require.ErrorContains(t, err, "opening source file")
+}
+
+func TestWriteRPMScript(t *testing.T) {
+	// Setup
+	_, _, teardown := setupRPMSourceDir(t, true)
```

Comment:
We can merge after getting rid of this call.

Thanks for addressing all comments!